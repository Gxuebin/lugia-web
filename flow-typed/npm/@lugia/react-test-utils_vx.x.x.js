// flow-typed signature: 08aba5f61ec37d26f554edaf1347d46d
// flow-typed version: <<STUB>>/@lugia/react-test-utils_v^1.0.0/flow_v0.74.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@lugia/react-test-utils'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@lugia/react-test-utils' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@lugia/react-test-utils/interface/react-test-utils.flow' {
  declare module.exports: any;
}

// Filename aliases
declare module '@lugia/react-test-utils/index' {
  declare module.exports: $Exports<'@lugia/react-test-utils'>;
}
declare module '@lugia/react-test-utils/index.js' {
  declare module.exports: $Exports<'@lugia/react-test-utils'>;
}
declare module '@lugia/react-test-utils/interface/react-test-utils.flow.js' {
  declare module.exports: $Exports<'@lugia/react-test-utils/interface/react-test-utils.flow'>;
}
