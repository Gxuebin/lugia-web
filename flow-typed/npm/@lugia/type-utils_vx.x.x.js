// flow-typed signature: 5896be137bd1cbb78f557576b3697b70
// flow-typed version: <<STUB>>/@lugia/type-utils_v^1.0.4/flow_v0.74.0

/**
 * This is an autogenerated libdef stub for:
 *
 *   '@lugia/type-utils'
 *
 * Fill this stub out by replacing all the `any` types.
 *
 * Once filled out, we encourage you to share your work with the
 * community by sending a pull request to:
 * https://github.com/flowtype/flow-typed
 */

declare module '@lugia/type-utils' {
  declare module.exports: any;
}

/**
 * We include stubs for each file inside this npm package in case you need to
 * require those files directly. Feel free to delete any files that aren't
 * needed.
 */
declare module '@lugia/type-utils/interface/type-utils.flow' {
  declare module.exports: any;
}

declare module '@lugia/type-utils/lib/ArrayUtils' {
  declare module.exports: any;
}

declare module '@lugia/type-utils/lib/index' {
  declare module.exports: any;
}

declare module '@lugia/type-utils/lib/NumberUtils' {
  declare module.exports: any;
}

declare module '@lugia/type-utils/lib/ObjectUtils' {
  declare module.exports: any;
}

declare module '@lugia/type-utils/lib/StringUtils' {
  declare module.exports: any;
}

// Filename aliases
declare module '@lugia/type-utils/interface/type-utils.flow.js' {
  declare module.exports: $Exports<'@lugia/type-utils/interface/type-utils.flow'>;
}
declare module '@lugia/type-utils/lib/ArrayUtils.js' {
  declare module.exports: $Exports<'@lugia/type-utils/lib/ArrayUtils'>;
}
declare module '@lugia/type-utils/lib/index.js' {
  declare module.exports: $Exports<'@lugia/type-utils/lib/index'>;
}
declare module '@lugia/type-utils/lib/NumberUtils.js' {
  declare module.exports: $Exports<'@lugia/type-utils/lib/NumberUtils'>;
}
declare module '@lugia/type-utils/lib/ObjectUtils.js' {
  declare module.exports: $Exports<'@lugia/type-utils/lib/ObjectUtils'>;
}
declare module '@lugia/type-utils/lib/StringUtils.js' {
  declare module.exports: $Exports<'@lugia/type-utils/lib/StringUtils'>;
}
