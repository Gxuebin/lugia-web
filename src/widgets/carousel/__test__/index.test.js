/**
 *
 * create by ligx
 *
 * @flow
 */
import React from 'react';

import 'jest-styled-components';
import Enzyme, { mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Carousel from '../carousel';
import renderer from 'react-test-renderer';
import { delay } from '@lugia/react-test-utils';

Enzyme.configure({ adapter: new Adapter() });

describe('Carousel', () => {
  it('getItemWrap', () => {
    const item = 'hello world';
    const result = Carousel.prototype.getItemWrap({
      switchType: 'horizontal',
      start: 0,
      index: 1,
      width: 50,
      height: 50,
      item,
    });

    expect(renderer.create(<div>{result}</div>).toJSON()).toMatchSnapshot();
  });
});
