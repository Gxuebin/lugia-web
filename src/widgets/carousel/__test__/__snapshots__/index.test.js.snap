// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel getItemWrap switchType : fade 1`] = `
.c0 {
  display: inline-block;
  position: absolute;
  opacity: 0;
  height: 4.166666666666667em;
  width: 4.166666666666667em;
  -webkit-transition: opacity 500s;
  transition: opacity 500s;
  overflow: hidden;
  vertical-align: top;
}

<div>
  <div
    checked={false}
    className="c0"
    height={50}
    width={50}
  >
    hello world
  </div>
</div>
`;

exports[`Carousel getItemWrap switchType : horizontal 1`] = `
.c0 {
  display: inline-block;
  height: 4.166666666666667em;
  width: 4.166666666666667em;
  -webkit-transition: opacity 500s;
  transition: opacity 500s;
  overflow: hidden;
  vertical-align: top;
}

<div>
  <div
    checked={false}
    className="c0"
    height={50}
    width={50}
  >
    hello world
  </div>
</div>
`;

exports[`Carousel getItemWrap switchType : vertical 1`] = `
.c0 {
  display: inline-block;
  height: 4.166666666666667em;
  width: 4.166666666666667em;
  -webkit-transition: opacity 500s;
  transition: opacity 500s;
  overflow: hidden;
  vertical-align: top;
}

<div>
  <div
    checked={false}
    className="c0"
    height={50}
    width={50}
  >
    hello world
  </div>
</div>
`;
